docs/multicluster/architecture.md

```diff
@@ -0,0 +1,95 @@
+# Antrea Multi-cluster Architecture
+
+Antrea Multi-cluster implements [Multi-cluster Service API](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api) for Service communication
+across multiple Kubernetes clusters.
+
+Below diagram depicts a basic multi-cluster topology in Antrea.
+
+<img src="assets/basic-topology.png" width="400" alt="Antrea Multi-cluster Topology">
+
+Given a set of Kubernetes clusters, there will be a leader cluster and several member clusters.
+By default, a leader cluster itself is also a member cluster of a ClusterSet. A cluster
+can also be configured as a dedicated leader cluster of multiple ClusterSets.
+
+## Terminology
+
+ClusterSet is a placeholder name for a group of clusters with a high degree of mutual
+trust and shared ownership that share Services amongst themselves. Within a ClusterSet,
+Namespace sameness applies, which means all Namespaces with a given name are considered to
+be the same Namespace. It's also a Custom Resource Definition which defines a ClusterSet
+including the leader and member clusters information.
+
+ClusterClaim is a Custom Resource Definition which can be used to define a cluster itself as
+a member of a ClusterSet with unique cluster ID, or claim a ClusterSet with unique ClusterSet
+ID.
+
+MemberClusterAnnounce is a Custom Resource Definition which declares a member cluster
+configuration to the leader cluster.
+
+Common Area is an abstraction in Multi-cluster implementation that provides interfaces to a
+persistent storage that can be read/write and shared for all member clusters and leader clusters.
+
+## Antrea Multi-cluster Controller
+
+In a member cluster, Antrea Multi-cluster creates a Deployment that runs Antrea Multi-cluster
+Controller which is responsible for exporting resource to and importing resource from a leader
+cluster in a ClusterSet.
+
+In a leader cluster, Antrea Multi-cluster creates a Deployment that runs Antrea Multi-cluster
+Controller which is responsible for converting resources from different member clusters into one
+encapsulated resource as long as these resources have the same kind and match Namespace sameness.
+
+In ClusterSet initialization, Antrea Multi-cluster Controller in a member cluster watches
+ClusterSet, ClusterClaim, and creates a MemberClusterAnnounce in the leader cluster.
+
+For resource exporting, it watches ServiceExport, Service and Endpoints resources from Kubernetes
+API, encapsulates Service and Endpoints into ResourceExport according to ServiceExport, and
+write the ResourceExport to leader cluster. For resource importing, it watches ResourceImport
+from leader cluster, and creates multi-cluster Service and Endpoints with a prefix `antrea-mc-`
+plus exported Service name, and also a ServiceImport which has the same name as the exported Service.
+
+In a leader cluster, For ClusterSet initialization, Antrea Multi-cluster controller watches
```

Comment:
Remove this paragraph? I think it is covered by the previous ones.

Reply:
it's for clusterset initialization in leader cluster, it's a little different between member and leader. so I suppose we should keep it.